### PR TITLE
Examples/cms-wordpress migrate to TypeScript

### LINKED ISSUE #38752: cms-wordpress typescript example
### Describe the feature you'd like to request

I'd like to use a typescript version of the wordpress cms example

### Describe the solution you'd like

A typescript version of the wordpress cms example should be added for ease of development.

### Describe alternatives you've considered

None.

### PR DESCRIPTION
**Pull request by chaiwattsw**
## Documentation / Examples

Not sure that I need to provide type to all of components or not?
Let me know if you want me to provide.
- [x] Closes #38752
- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/cms-wordpress/next.config.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ if (!process.env.WORDPRESS_API_URL) {
 module.exports = {
   images: {
     domains: [
-      process.env.WORDPRESS_API_URL.match(/(http(?:s)?:\/\/)(.*)/)[2], // Valid WP Image domain.
+      process.env.WORDPRESS_API_URL.match(/http(?:s)?:\/\/(.[^/]+)/)[1], // Valid WP Image domain.
```
Comment (leerob) [inline examples/cms-wordpress/next.config.js:11]: Why did this change?

Comment (chaiwattsw) [reply examples/cms-wordpress/next.config.js]: For example if we have `WORDPRESS_API_URL  = "http://127.0.0.1/wordpress/graphql"`
When we using `match(/http(?:s)?:\/\/(.[^/]+)/)` 

## **Actual Result**

`WORDPRESS_API_URL.match(/http(?:s)?:\/\/(.[^/]+)/)[2] // 127.0.0.1/wordpress/graphql` 
Which is not valid WP image domain so `Next` will throw error on `<Image/>` component
exciting-mcnulty-bi5ylq - CodeSandbox - Google Chrome_2

## **Expected Result**  

- 127.0.0.1
- example.com `(if WORDPRESS_API_URL = "http://example.com/wordpress/graphql" or "http://www.example.com/wordpress/graphql")` currently this regex not extract www from url will fix this in next commit

**Review by leerob — APPROVED** [review]
Thank you!

**Comment by chaiwattsw** [comment]
> Thank you!

@leerob My pleasure. Could you approve workflow to run again please

**Review by leerob — APPROVED** [review]

**Comment by chaiwattsw** [comment]
Still can't automatically merge, That's weird.

**Review by ijjk — APPROVED** [review]

**Pull request merged**